src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedContainer.cs

```diff
@@ -123,16 +125,40 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
  
         public override ImmutableArray<Symbol> GetMembers()
         {
+            var builder = ArrayBuilder<Symbol>.GetInstance();
+            if (this.TypeKind == TypeKind.Struct)
+            {
+                // Add an extra object field for EnC.
+                builder.Add(GetStructFieldCaptureObject());
+            }
+
             Symbol constructor = this.Constructor;
-            return (object)constructor == null ? ImmutableArray<Symbol>.Empty : ImmutableArray.Create(constructor);
+            if ((object)constructor != null)
+            {
+                builder.Add(constructor);
+            }
+
+            return builder.ToImmutableAndFree();
         }
 
         public override ImmutableArray<Symbol> GetMembers(string name)
         {
```

Comment:
Agree. Good idea! Thanks!

---
In reply to: [134591158](https://github.com/dotnet/roslyn/pull/21664#discussion_r134591158) [](ancestors = 134591158)